### PR TITLE
Bug 1380408 - elasticsearch user completion should consider people with multiple ircnicks

### PR DESCRIPTION
**Pull request by dylanwh**

### CODE REVIEW

File: Bugzilla/User.pm [hunk 1]
```diff
@@ -176,21 +176,28 @@ sub ES_PROPERTIES {
...
         suggest_user => {
             input => [ $self->login, $self->name ],
             output => $self->identity,
             payload => { name => $self->login, real_name => $self->name },
         },
     };
-    if ($self->name && $self->name =~ /:(\w+)/) {
-        my $ircnick = $1;
+    my @nicks;
+    if ($self->name) {
```
Comment (floatingatoll) [inline Bugzilla/User.pm]: What is this if() testing for? When would 'if ($self->name)' NOT product a warning, but then $name =~ /:.../ would?

Comment (floatingatoll) [inline Bugzilla/User.pm]: `my(@nicks) = ($self->name =~ m//mg);` being the alternate form I envisioned for 188...

File: Bugzilla/Elastic/Indexer.pm [hunk 2]
```diff
@@ -179,6 +179,7 @@ sub bulk_load {
     my $new_ids     = $self->_select_all_ids($class, $last_id);
     my $updated_ids = $self->_select_updated_ids($class, $last_mtime);
 
+    $self->put_mapping($class);
```
Comment (dylanwh) [inline Bugzilla/Elastic/Indexer.pm]: This is actually fixing a pretty bad mistake in the old code, that wouldn't update the mapping if we ever deleted the ES index.

Comment (floatingatoll) [inline Bugzilla/Elastic/Indexer.pm]: Has it been submitted upstream, and/or is it relevant to anyone other than BMO?

**Pull request merged**